cloud-init.pl

```diff
@@ -84,6 +84,25 @@ sub apply_user_data {
       system("sh -c \"$runcmd\"");
     }
   }
+
+  if (defined($data->{write_files})) {
+    foreach my $item (@{ $data->{write_files} }) {
+      open my $fh, ">", $item->{path};
```

Comment:
It might be worth doing an `mkdir -p` on the path's dirname before attempting to write.
 Something along the lines of:

```perl
use File::Basename;
use File::Path;

[...]

mkpath [dirname($item)], 1, 0755;
```

Reply:
Agreed, will test & update the PR